logic/transaction.js

```diff
@@ -407,15 +407,24 @@ Transaction.prototype.process = function (trs, sender, requester, cb) {
  * @param {transaction} trs
  * @param {account} sender
  * @param {account} requester
+ * @param {boolean} checkConfirmed - Check if trs already persisted in database
  * @param {function} cb
  * @return {setImmediateCallback} validation errors | trs
  */
-Transaction.prototype.verify = function (trs, sender, requester, cb) {
+Transaction.prototype.verify = function (trs, sender, requester, checkConfirmed, cb) {
```

Comment:
Please refactor all calls of that function to always pass all parameters instead of later `if (typeof requester === 'function') {` thing.